io/ilvis2/Ilvis2Reader.cpp

```diff
@@ -123,11 +125,83 @@ T convert(const StringList& s, const std::string& name, size_t fieldno)
 }
 
 
+// If longitude between 0-180, just return it, degrees east; if between 180
+// and 360, subtract 360 to get negative value.
+double Ilvis2Reader::convertLongitude(double longitude)
+{
+  if (longitude > 180.0)
+  {
+    return longitude - 360.0;
+  }
+  else
+  {
+    return longitude;
+  }
+}
+
+
+void Ilvis2Reader::readPoint(PointViewPtr view, PointId nextId, StringList s, std::string pointMap)
+{
+  PointLayoutPtr layout = view->layout();
+
+  std::string name("LVIS_LFID");
```

Comment:
I know that you're just moving this, but there's no reason I can think of not to make these real dimensions (add to Dimension.hpp).  It gets away from all the string stuff, dealing with the layout and the dimension finding.  It'll let you just say `setField(Dimension::Id::LongitudeCentroid,...)